src/mappings/euler.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  Address,
+  BigInt,
+  Bytes,
+  crypto,
+  ethereum,
+  store,
+  log
+} from '@graphprotocol/graph-ts'
+
+import {
+  PoolDeployed as HookDeployedEvent,
+  PoolUninstalled as HookUninstalledEvent,
+  PoolConfig as HookConfigEvent,
+} from '../types/EulerSwapFactory/EulerSwapFactory'
+import { EulerSwapHook } from '../types/schema'
+
+
+export function handleHookDeployed(event: HookDeployedEvent): void {
+  // ──────────────────────────────────────────────────────────────
+  // Build the composite ID: eulerAccount-asset0-asset1
+  // ──────────────────────────────────────────────────────────────
+  const account = event.params.eulerAccount.toHexString()
+  const asset0  = event.params.asset0.toHexString()
+  const asset1  = event.params.asset1.toHexString()
+
+  const id = `${account}-${asset0}-${asset1}`          // <- entity id
+
+  // ──────────────────────────────────────────────────────────────
+  // Load (or create) the row for this tuple
+  // ──────────────────────────────────────────────────────────────
+  let entity = EulerSwapHook.load(id)
+  if (entity == null) {
+    // This is a logic error → stop indexing so we notice immediately
+    log.critical('PoolConfig emitted before PoolDeployed for id {}', [id])
+    return 
+  }
+
+  // Always overwrite the hook address so the latest hook is used
+  entity.hook = event.params.pool.toHexString()
+
+  entity.save()
+}
+
+export function handleHookUninstalled(event: HookUninstalledEvent): void {
+  // ──────────────────────────────────────────────────────────────
+  // Build the composite ID: eulerAccount-asset0-asset1
+  // ──────────────────────────────────────────────────────────────
+  const account = event.params.eulerAccount.toHexString()
+  const asset0  = event.params.asset0.toHexString()
+  const asset1  = event.params.asset1.toHexString()
+
+  const id = `${account}-${asset0}-${asset1}`          // <- entity id
+
+  // ──────────────────────────────────────────────────────────────
+  // Load (or create) the row for this tuple
+  // ──────────────────────────────────────────────────────────────
+  let entity = EulerSwapHook.load(id)
+  if (entity != null && entity.hook == event.params.pool.toHexString()) {
+    store.remove("EulerSwapHook", id)
+  }
+}
+
+// ───────────────────────────────────────────────────────────────
+// NEW: PoolConfig — derive poolId from PoolKey
+// ───────────────────────────────────────────────────────────────
+export function handlePoolConfig(event: HookConfigEvent): void {
+  // params struct fields
+  const p          = event.params.params
+  const account    = p.eulerAccount.toHexString()
+  const asset0     = p.currency0.toHexString()            // ↳ names may differ
+  const asset1     = p.currency1.toHexString()
```

Comment:
@david-defi-uf Thanks for the PR! Im trying to build and deploy with these changes but looks like `currency0` and `currency1` are not part of the params [tuple](https://github.com/Uniswap/v4-subgraph/blob/3d5ba66278c1d8c5497044a1cb76d7c9dd268771/abis/EulerSwapFactory.json#L38-L50). I see `vault0` and `vault1` but those seems to be different than the tokens of the pool if I understand correctly.